adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Exceptions/AdalExceptionService.cs

```diff
@@ -0,0 +1,139 @@
+﻿//----------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+using Microsoft.Identity.Core;
+using System;
+using System.Globalization;
+using System.Text;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    /// <summary>
+    /// Implementation of the <see cref="CoreExceptionService"/> that throws <see cref="AdalException"/>
+    /// </summary>
+    /// <remarks>Does not currently throw <see cref="AdalSilentTokenAcquisitionException"/> and 
```

Comment:
What does this remark mean? Can you please elaborate (I'm confused if this is a good thing, a bad thing, more work is required ?)

Reply:
from related PR, this is the placeholder for ADAL exceptions ensuring we follow the same path for both libraries. I assume it means those exceptions are still thrown in place and not using this structure.

---
In reply to: [205396669](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1148#discussion_r205396669) [](ancestors = 205396669)

Reply:
Yes, exactly, I am not sure we'll ever throw those exceptions from core. Both adal and msal still throw exceptions using `throw new AdalServiceException` and do not use the factories.